### PR TITLE
Provide stores to RemoteDataServicesManager directly

### PR DESCRIPTION
**Pull request by ps2**
https://tidepool.atlassian.net/browse/LOOP-969

### CODE REVIEW

File: Loop/Managers/DeviceDataManager.swift [hunk 1]
```diff
@@ -66,7 +66,7 @@ final class DeviceDataManager {
 
     private(set) var servicesManager: ServicesManager!
 
-    var analyticsServicesManager: AnalyticsServicesManager { return servicesManager.analyticsServicesManager }
+    var analyticsServicesManager: AnalyticsServicesManager
 
     var loggingServicesManager: LoggingServicesManager { return servicesManager.loggingServicesManager }
```
Comment (nhamming) [inline Loop/Managers/DeviceDataManager.swift]: Curious why `analyticsServicesManager` gets special treatment, but `loggingServicesManager` and `remoteDataServicesManager` still reach into another manager?

**Review by nhamming — COMMENTED** [review]
Question about consistency.

File: Loop/Managers/ServicesManager.swift [hunk 2]
```diff
@@ -23,13 +23,16 @@ class ServicesManager {
 
     private let servicesLock = UnfairLock()
 
-    init(pluginManager: PluginManager, deviceDataManager: DeviceDataManager) {
+    init(
```
Comment (darinkrauss) [inline Loop/Managers/ServicesManager.swift:26]: If you are going to pass in both `remoteDataServicesManager` and `analyticsServicesManger` why not `loggingServicesManager`, too?

File: Loop/Managers/ServicesManager.swift [hunk 3]
```diff
@@ -23,13 +23,16 @@ class ServicesManager {
...
+        pluginManager: PluginManager,
+        deviceDataManager: DeviceDataManager,
+        remoteDataServicesManager: RemoteDataServicesManager,
```
Comment (darinkrauss) [inline Loop/Managers/ServicesManager.swift]: [nit] I'd order the parameters to match the declarations above (i.e. `analyticsServicesManager` then `loggingServicesManager` then `remoteDataServicesManager`).

File: Loop/Managers/DeviceDataManager.swift (hunk 1 above)
Comment (ps2) [reply Loop/Managers/DeviceDataManager.swift]: Those are still tied up in a dependency cycle of DDM -> LDM -> stores -> RDM.  There is a bunch more refactoring that needs to happen here, but I'm holding off, as there are too many in-flight changes between DIY (significant changes in LDM) and Tidepool (significant changes in DDM) atm.  I am targeting the next refactoring for after the next sync point.

Comment (darinkrauss) [reply Loop/Managers/DeviceDataManager.swift]: Actually, I think `loggingServicesManager` should be created and owned by `DeviceDataManager` just like `analyticsServicesManager`.

**Review by darinkrauss — CHANGES_REQUESTED** [review]
Please see comments about creation/ownership of `loggingServicesManager`.

File: Loop/Managers/DeviceDataManager.swift (hunk 1 above)
Comment (ps2) [reply Loop/Managers/DeviceDataManager.swift]: Oh, yeah, LoginServicesManager is not dependent on the stores either. Changing this. Thanks.

**Review by darinkrauss — APPROVED** [review]
LGTM 🚢

**Review by nhamming — APPROVED** [review]
LGTM

**Pull request merged**